canvas_api_client/canvas_api_client/v1_client.py

```diff
@@ -141,17 +153,50 @@ def _get_paginated(self,
                 response.links['next']['url'], headers=headers)
             yield response.json()
 
-    def _format_sis_course_id(self, course_id: str):
+    def _get_flattened(self,
+                       url: str,
+                       headers: RequestHeaders = None,
+                       params: RequestParams = None) -> Iterator[Response]:
+        """
+        Send an API call to the Canvas server with pagination.
+
+        Returns a generator of response objects.
+        """
+        response = self._get(url, headers=headers, params=params)
+        self._check_response_headers_for_pagination(response)
+
+        for item in response.json():
+            yield item
+
+        while 'next' in response.links:
+            response = self._get(
+                response.links['next']['url'],
+                headers=headers
+                )
+
+            for item in response.json():
+                yield item
```

Comment:
Why can't we just call `_get_paginated` in this method and flatten that with `itertools.chain.from_iterable`?

Reply:
Because the code is simple enough to do without `itertools`, and it's not clear what `itertools` adds to it?

Reply:
Because the code is simple enough to do without itertools, and it's not clear what itertools adds to it?

Reply:
it would reduce 10 lines of code to 3...

Reply:
and `itertools` is a built-in, it's not like we need to install an extra dependency. less code we have to maintain the better imo.

Reply:
I guess I'm uncomfortable creating a complex data structure - which could get pretty large - only to flatten it.  Why not create the flat data structure from the beginning?

Reply:
For me I think it's really important for an API client to provide the user with exactly what the API client returns. That's why I think if the default response is paginated, our default behavior should be to also return a paginated representation of the data. I'm comfortable adding helping methods for flattening the data structure, but I don't think that should be the default. 

I'm actually going to step back a bit on using `itertools` here. I was misunderstanding the data structure yesterday, I think this is actually pretty nice. However I do stand by the principle that the API client should provide the user with what the API returns.

Reply:
Sure, but (a) if you believe in private methods, this is one, and (b) we have the `_get_paginated()` method that still does that.

Reply:
Are you arguing that the response will not be paginated by default? Otherwise I am fine.

Reply:
Like I've said from the beginning, I would be fine passing a keyword arg and an arg to the class instance that overrides this.

Reply:
The way I've written this is that if we call `_get_paginated()` we'll get a paginated result.  If we call `_get_flattened()` we'll get a flattened result.

Reply:
I see that now, and this looks good! Feel free to resolve this.